src/runtime/plugins/supabase.client.ts

```diff
@@ -46,3 +49,60 @@ const setServerSession = (event: AuthChangeEvent, session: Session | null) => {
     body: { event, session }
   })
 }
+
+
+const setClientSession = (event: AuthChangeEvent, session: Session | null) => {
+  // https://nuxt.com/docs/api/composables/use-cookie
+  
+  const config = useRuntimeConfig().public
+  const cookieOptions = config.supabase.cookies
+  
+  if (event === 'SIGNED_IN' || event === 'TOKEN_REFRESHED') {
+    console.info('not setServerSession create cookie', event)
+    if (!session) { throw new Error('Auth session missing!') }
+
+    const cookie1 = useCookie(
+      `${cookieOptions.name}-access-token`,
+      {
+        domain: cookieOptions.domain,
+        maxAge: cookieOptions.lifetime ?? 0,
+        path: cookieOptions.path,
+        sameSite: cookieOptions.sameSite as boolean | 'lax' | 'strict' | 'none'
+      })
+    cookie1.value = session.access_token
+    console.info('cookie1', cookie1)
+
+    const cookie2 = useCookie(
+      `${cookieOptions.name}-refresh-token`,
+      {
+        domain: cookieOptions.domain,
+        maxAge: cookieOptions.lifetime ?? 0,
+        path: cookieOptions.path,
+        sameSite: cookieOptions.sameSite as boolean | 'lax' | 'strict' | 'none'
+      })
+    cookie2.value = session.refresh_token
+    console.info('cookie2', cookie2)
+  }
+
+  if (event === 'SIGNED_OUT') {
+    console.info('not setServerSession clear cookie', event)
+
+    const cookie1 = useCookie(
+      `${cookieOptions.name}-access-token`,
+      {
+        maxAge: -1,
+        path: cookieOptions.path
+      })
+    cookie1.value = ''
+
+    const cookie2 = useCookie(
+      `${cookieOptions.name}-refresh-token`,
+      {
+        maxAge: -1,
+        path: cookieOptions.path
+      })
+    cookie2.value = ''
+
+  }
```

Comment:
Since this code is a duplicate of [this one](https://github.com/nuxt-modules/supabase/blob/main/src/runtime/server/api/session.ts#L9), I think it should be mutualised in a `setSupabaseCookies` in a new utility file: `src/utils/cookies.ts`. This method could then be called from both `plugins/supabase.client.ts` and `api/session.ts`.

Reply:
You're right. I will do that.

Could you please give me some instructions on how to setup my dev/test env ?
I [read the dev section](https://github.com/nuxt-modules/supabase#development) but since we need a static build I think the playground is not suitable for this case.

How would you set up pnpm to develop on this project while you test it from another project ?
I found the `pnpm link` command but I am a bit confused with it :cry:

Reply:
I dont know the difference between `useCookie()` and `setCookie()`. Shall we replace `setCookie` by `useCookie` if it can run both on client and server sides ? The [nuxt doc is not really clear](https://nuxt.com/docs/api/composables/use-cookie#handling-cookies-in-api-routes) on that.

Reply:
I'm not sure but `useCookie` could be wrapper around the `getCookie` and `setCookie` of `unjs/h3`. It means you should be able to use `getCookie` and `setCookie` in both client and server context. Let's try it.

Reply:
I think you can at least play with the playground by using the `ssr: false` option. I'm using a classical `yarn link` to use the module in another project (if you want to try with a static build).